pkg/clustermesh/remote_cluster.go

```diff
@@ -251,6 +254,39 @@ func (rc *remoteCluster) onInsert(allocator RemoteIdentityWatcher) {
 			}
 		}
 	}()
+
+	rc.controllers.UpdateController(rc.remoteConnectionControllerName+"-status",
+		controller.ControllerParams{
+			DoFunc: func(ctx context.Context) error {
+				for {
+					select {
+					// terminate routine when remote cluster is removed
+					case _, ok := <-rc.changed:
+						if !ok {
+							return nil
+						}
+					default:
+					}
+
+					// wait for backend to appear
+					rc.mutex.RLock()
+					if rc.backend == nil {
+						rc.mutex.RUnlock()
+						time.Sleep(10 * time.Millisecond)
+						continue
+					}
+					statusCheckErrors := rc.backend.StatusCheckErrors()
+					rc.mutex.RUnlock()
+
+					err, ok := <-statusCheckErrors
+					if ok && err != nil {
+						rc.getLogger().WithError(err).Warning("Error observed on etcd connection, reconnecting etcd")
```

Comment:
At the moment this is hiding the errors from the controller infrastructure, which means it's no more visible than before; we'll just see yet another controller running with an "OK" status. Could/should we return an error in this case to get the controller infrastructure to retrigger this logic? Or are we trying to avoid it to keep retries to a short period?

Reply:
I've had to remove the controller again. The controller architecture is not yet useful to listen on channels as it relies on completing runs regularly in order to terminate controllers. I've switched back to a go routine to make backports feasible.

Reply:
I wondered if that might be the conclusion, wasn't sure if we can get away with the controller not triggering for a long time then it will just reschedule once we return an error. But if we need more controller package infrastructure then it makes sense to go back to how it was. Thanks for exploring the idea anyway.